### PR TITLE
Merged current main into testing branch

### PR DESCRIPTION
**Pull request by NicMilli**

### CODE REVIEW

File: client/src/components/ProductDetails/Details.jsx
Comment (B-Tanonaka) [inline client/src/components/ProductDetails/Details.jsx:1]: Favoriting an item does not actively change the heart to red but does on refresh.
Not a blocker, just a heads up.

File: client/src/features/api/apiSlice.js
Comment (B-Tanonaka) [inline client/src/features/api/apiSlice.js:1]: Nothing changed, just moved lines around to be more readable.

**Review by B-Tanonaka — COMMENTED** [review]
Prior to merging with current relatedItems-v3.

I checked through everything except my own components.

**Review by msong1 — COMMENTED** [review]
- All the tests pass
- The test coverage for lines is more than 73.57%

**Review by msong1 — APPROVED** [review]
- All the tests pass
- The test coverage for lines is more than 73.57%
- The page loads well and CSS style changes seem to work.
Great job!

**Pull request merged**